### PR TITLE
Use state-backed definition loader in Airlift

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

Example application of StateBackedDefinitionsLoader in Airlift

## How I Tested These Changes

BK

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by schrockn** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24647?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24647** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24647?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24646** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24646?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @schrockn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by schrockn — COMMENTED** [review]
Found bug. Might be in core framework backing this. Investigating.

File: examples/experimental/dagster-airlift/dagster_airlift/core/load_defs.py [hunk 1]
```diff
@@ -19,26 +18,33 @@
...
-            context.reconstruction_metadata[metadata_key], as_type=AirflowDefinitionsData
-        )
-        assert isinstance(airflow_definition_data, AirflowDefinitionsData)
-        return airflow_definition_data
-    else:
-        serialized_data = compute_serialized_data(airflow_instance=airflow_instance, defs=defs)
+@dataclass
+class AirflowInstanceDefsLoader(StateBackedDefinitionsLoader[AirflowDefinitionsData]):
+    airflow_instance: AirflowInstance
+    explicit_defs: Definitions
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/load_defs.py:27]: @dpeng817 want to call out this terminology change. Kept on getting the defs passed in and the defs programmatically generated so moved this to `explicit_defs` to indicate these are the ones that the user passed in.

**Review by dpeng817 — APPROVED** [review]
nice.

**Comment by schrockn** [comment]
### Merge activity

* **Sep 24, 4:32 PM EDT**: @schrockn started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/24647).

**Pull request merged**